### DISCUSSION TITLE
Can I restart the network stack?

### DISCUSSION THREAD

**Discussion by jeffer80** [nanoframework/Home#1355] [Q&A] [answered]
In my project, I need to configure my ESP32 as an AP via Bluetooth and I found the following sample code.
The problem is device must be rebooted after the configuration is written, and I'll last my Bluetooth connection.
Can I restart the network stack only instead of rebooting the whole device?

[https://github.com/nanoframework/Samples/blob/main/samples/WiFiAP/Program.cs]
```
if (WirelessAP.Setup() == false)
{
  // Reboot device to Activate Access Point on restart
  Debug.WriteLine($"Setup Soft AP, Rebooting device");
  Power.RebootDevice();
}
```

**Comment by Ellerbach** [comment]
The way IDF works, is yes, to reboot the device to reboot the network stack on ESP32 familly.

**Reply by jeffer80** [reply]
But rebooting device will disconnect the bluetooth connection, can I only restart the network stack without affecting the BT or other tasks.

**Reply by josesimoes** [reply, marked as answer]
Understood that you want to restart only the network stack. With the current implementation that is not possible.
This is because of the high intricacy on the IDF lwIP stack and nanoFramework network driver. Achieving a level of decoupling to accomplish your requirement would need a serious refactoring.

**Reply by jeffer80** [reply]
thx, maybe I should review my requirement and change the implementation flow
